doc/internals/contributing.rst

```diff
@@ -0,0 +1,309 @@
+======================
```

Comment:
Could you separate the commits into moving (renaming) and modification? AFAIK, It keeps the change history of the file. Additionally, it helps to read changes from individual commit logs for me.

I know it does not help file separation. But it would be better to keep history even only for "contributing to sphinx" part.

Reply:
I've tried to do this. Hopefully it's okay now?

Reply:
No, it's not good. It seems you did moving CONTRIBUTING.rst to doc/internals/contributing.rst and adding new CONTRIBUTING.rst in the same commit. It seems git considers it large deletion of CONTRIBUTING.rst and creation of doc/internals/contributing.rst.
https://github.com/sphinx-doc/sphinx/pull/7750/commits/7e8ae1127577ca60d6ce98b9c958b6ddb554a44e

As a result, the history of doc/internals/contributing.rst has been cleared.
https://github.com/stephenfin/sphinx/blame/doc/internals-section/doc/internals/contributing.rst


Please separate renaming and other actions.
For example, this branch commits 3 times; 1) moving 2) adding 3) modifying. It allows to keep history.
https://github.com/sphinx-doc/sphinx/compare/3.x...tk0miya:moving_test
https://github.com/tk0miya/sphinx/blame/moving_test/doc/internals/contributing.rst

Reply:
Okay, I think I have this done now.